In [1]:
import pandas as pd
import numpy as np
import seaborn as sna
import matplotlib.pyplot as plt
from scipy import stats
from scipy.stats import chi2_contingency
import warnings
warnings.filterwarnings('ignore')

# Que no.1
- A F&B manager wants to determine whether there is any     significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.


- 
- Null hypothesis : cutlets size is same
- Alternative Hypothesis : Cutlets size are different

In [2]:
cutlet =pd.read_csv("Cutlets.csv")
cutlet

Unit A  Unit B
0   6.8090  6.7703
1   6.4376  7.5093
2   6.9157  6.7300
3   7.3012  6.7878
4   7.4488  7.1522
5   7.3871  6.8110
6   6.8755  7.2212
7   7.0621  6.6606
8   6.6840  7.2402
9   6.8236  7.0503
10  7.3930  6.8810
11  7.5169  7.4059
12  6.9246  6.7652
13  6.9256  6.0380
14  6.5797  7.1581
15  6.8394  7.0240
16  6.5970  6.6672
17  7.2705  7.4314
18  7.2828  7.3070
19  7.3495  6.7478
20  6.9438  6.8889
21  7.1560  7.4220
22  6.5341  6.5217
23  7.2854  7.1688
24  6.9952  6.7594
25  6.8568  6.9399
26  7.2163  7.0133
27  6.6801  6.9182
28  6.9431  6.3346
29  7.0852  7.5459
30  6.7794  7.0992
31  7.2783  7.1180
32  7.1561  6.6965
33  7.3943  6.5780
34  6.9405  7.3875

In [3]:
stats.ttest_ind(cutlet["Unit A"],cutlet["Unit B"])

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

- p-value is greater than alpha value.Accept Null hypothesis
- The cutlets size is same

# Question 02
-    A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch.
   
- Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.

- Null Hypothesis: all sample is same
- Alternative Hypothesis : At least one sample is different

In [4]:
from scipy.stats import stats as st

In [5]:
lab = pd.read_csv("LabTAT.csv")

In [6]:
lab.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [7]:
lab.iloc[:,0]

0      185.35
1      170.49
2      192.77
3      177.33
4      193.41
        ...  
115    178.49
116    176.08
117    202.48
118    182.40
119    182.09
Name: Laboratory 1, Length: 120, dtype: float64

In [8]:
st.f_oneway(lab.iloc[:,0],lab.iloc[:,1],lab.iloc[:,2],lab.iloc[:,3])

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

- p value is less then alpha value
- Rejet Null hypothesis and Accept Alternate hypothesis

# Question 03
- Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.

- Null Hypothesis: all proportions are equal
- Alternative Hypothesis : Not all Proportion are equal

In [9]:
df = pd.read_csv("BuyerRatio.csv")
df

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [10]:
mf = np.array([[50,142,131,70],[435,1523,1356,750]])
mf

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]])

In [11]:
chi,p,df,expected = chi2_contingency(mf)

In [12]:
chi

1.595945538661058

In [13]:
p

0.6603094907091882

In [14]:
df

3

In [15]:
expected

array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]])

- 

- p value = 0.6603 > alpha value = 0.05
- Accept Null hypothesis
- all proportions are equal

# Question 04
- TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences

- Null Hypothesis : customer order forms defective % does not varies by center
- Alternative Hypothesis : varies by center

In [16]:
order = pd.read_csv("Costomer+OrderForm.csv")

In [17]:
order

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [18]:
order.value_counts().sum()

300

In [19]:
order.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [20]:
order.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [21]:
order.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [22]:
order.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [23]:
order1=np.array([[271,267,269,280],[29,33,31,20]])
order1

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [24]:
chi,p,df,expected = chi2_contingency(order1)

In [25]:
chi

3.858960685820355

In [26]:
p

0.2771020991233135

In [27]:
df

3

In [28]:
expected

array([[271.75, 271.75, 271.75, 271.75],
       [ 28.25,  28.25,  28.25,  28.25]])

- p value = 0.277 > alpha value = 0.05
- Accept Null Hypothesis
- customer order forms defective % does not varies by center.